src/glum/_util.py

```diff
@@ -53,6 +53,40 @@ def _align_df_categories(df, dtypes) -> pd.DataFrame:
     return df
 
 
+def _add_missing_categories(
+    df,
+    dtypes,
+    feature_names: Sequence[str],
+    categorical_format: str,
+    cat_missing_name: str,
+) -> pd.DataFrame:
+    if not isinstance(df, pd.DataFrame):
+        raise TypeError(f"Expected `pandas.DataFrame'; got {type(df)}.")
+
+    changed_dtypes = {}
+
+    categorical_dtypes = [
+        column
+        for column, dtype in dtypes.items()
+        if pd.api.types.is_categorical_dtype(dtype) and (column in df)
```

Comment:
Are there cases in which it makes sense to fit or predict on a df that does not have all features in `self.feature_dtypes_,`? If we can't think of any, we should raise if the column cannot be found in df. We can always relax this later in case that a use case comes up.

Reply:
Good point. I was basing it off `_align_df_categories` which has the same logic. It might make sense to raise in both functions, though. Or, maybe, it could be part of a more general check (`df` has the expected columns in the expected order) in a separate function.

Reply:
@MatthiasSchmidtblaicherQC let me know if you are okay with doing it in a future PR (as the idea is somewhat more general than categorical missings IMO)

Reply:
Yeah sounds good. In case you don't make a PR until end of next week, can you open an issue?